### PR TITLE
Fix warnings that are causing build fails with specific build flags

### PR DESCRIPTION
**Pull request by enginmanap**
These changes are all minor changes that are aimed to fix warnings.

1) -Werror=ignored-qualifiers
2) -Werror=implicit-fallthrough
3) -Werror=deprecated-copy
4) -Werror=non-virtual-dtor

I am getting them with the following settings:
-std=c++14 -Wall -Wextra -Wnon-virtual-dtor -pedantic

I searched for a coding guide to verify these flags are valid for Assimp project, but couldn't find anything. As far as I can think of, they should not cause any negative effects, and all should be net positives, but if I missed something please let me know.

### CODE REVIEW

**Comment by enginmanap** [comment]
Hey there,

I added 2 commits, they were meant to be seperate PRs, but they somehow got in to this one.
These are:
1) don't force -03 on debug builds, instead use -g.
2) Fix collada import fails on Mixamo models. This one is cause by a leading empty space on vcount tag, which is read as 0. Fixes #4551

If combined PRs are not accepted, I can try to seperate them, please let me know.

File: code/AssetLib/MMD/MMDPmxParser.h [hunk 1]
```diff
@@ -357,6 +357,10 @@ namespace pmx
 	{
 	public:
 		void virtual Read(std::istream *stream, PmxSetting *setting) = 0;
+
+        virtual ~PmxMorphOffset() {
```
Comment (kimkulling) [inline code/AssetLib/MMD/MMDPmxParser.h:361]: Shall be default destructor.

**Review by kimkulling — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution. There are some small findings. I will merge your PR when their are done!

File: code/AssetLib/Collada/ColladaParser.cpp [hunk 2]
```diff
@@ -1616,12 +1616,17 @@ void ColladaParser::ReadIndexData(XmlNode &node, Mesh &pMesh) {
...
+                    SkipSpacesAndLineEnd(&content);
                     for (unsigned int a = 0; a < numPrimitives; a++) {
                         if (*content == 0) {
                             throw DeadlyImportError("Expected more values while reading <vcount> contents.");
                         }
                         // read a number
-                        vcount.push_back((size_t)strtoul10(content, &content));
+                        size_t valueRead = (size_t)strtoul10(content, &content);
+                        if(valueRead == 0) {
+                            printf("wohoo");
```
[the remark this answers is not in the document]
Comment (enginmanap) [reply code/AssetLib/Collada/ColladaParser.cpp]: it is, I missed it.

File: code/AssetLib/MMD/MMDPmxParser.h (hunk 1 above)
Comment (enginmanap) [reply code/AssetLib/MMD/MMDPmxParser.h]: its fixed now.

**Comment by enginmanap** [comment]
This change also fixes #4363

I assume same goes for 4535, but there is no model in that issue to test, so can't verify.

**Comment by enginmanap** [comment]
Hey @kimkulling , can you take another look at the PR?

**Comment by jinfagang** [comment]
Please merge this PR, it very useful.

**Review by kimkulling — APPROVED** [review]
Looks fine!

**Comment by kimkulling** [comment]
Merged, thanks for your contribution.

**Pull request merged**